core/src/main/java/org/sonarsource/sonarlint/core/WsHelperImpl.java

```diff
@@ -85,4 +93,62 @@ public String generateAuthenticationToken(ServerConfiguration serverConfig, Stri
     SonarLintWsClient client = createClient(serverConfig);
     return generateAuthenticationToken(new ServerVersionAndStatusChecker(client), client, name, force);
   }
+
+  @Override
+  public List<RemoteOrganization> listOrganizations(ServerConfiguration serverConfig) {
+    SonarLintWsClient client = createClient(serverConfig);
+    ServerVersionAndStatusChecker serverChecker = new ServerVersionAndStatusChecker(client);
+    return listOrganizations(client, serverChecker);
+  }
+
+  static List<RemoteOrganization> listOrganizations(SonarLintWsClient client, ServerVersionAndStatusChecker serverChecker) {
+    try {
+      serverChecker.checkVersionAndStatus("6.4");
+    } catch (UnsupportedServerException e) {
+      // Organization support added in 6.4
+      return Collections.emptyList();
+    }
+    List<RemoteOrganization> result = new ArrayList<>();
+    fetchOrganizations(client, result);
+    return result;
+  }
+
+  private static void fetchOrganizations(SonarLintWsClient client, List<RemoteOrganization> result) {
+    int page = 1;
+    int pageSize = 500;
+    int loaded = 0;
+
+    while (true) {
+      Organizations.SearchWsResponse response = loadFromStream(client.get(getUrl(page, pageSize)));
+      // FIXME waiting for proper pagination support, iterate until no results are returned (it cost one useless call)
+      if (response.getOrganizationsCount() == 0) {
+        break;
+      }
+      for (Organization org : response.getOrganizationsList()) {
+        result.add(new DefaultRemoteOrganization(org));
+      }
+      // loaded += response.getPs();
+      //
+      // if (response.getTotal() <= loaded) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)